### PR TITLE
Add protocol-based distribution conversion and KDEDistribution

### LINKED ISSUE #95: Distribution conversion automation
Currently there is no automatic conversion if a distribution doesn't support a needed functionality (that is, a specific Protocol). A canonical example is a discrete approximation (such as from MCMC) is available but SupportLogProb is needed. The distribution conversion registry should provide a way to convert a distribution without SupportsLogProb into one that satisfies SupportsLogProb. This should be doable with minimal changes. The crux is to include canonical / default approaches for converting between SupportsSampling and SupportsLogProb, perhaps with more specialized methods in certain cases, such as for EmpiricalDistribution or when there is support for computing expectations or means/variances. 

Example:
```python
# MCMC approximation 
posterior = condition_on(my_model, data)
# automatic conversion because prior must inherit SupportsLogProb
second_model = SimpleLogProbModel(likelihood=likelihood, prior=posterior) 
# uses NUTS be default 
second_posterior = condition_on(second_model, other_data)

# no conversion if only sampling needed 
alt_second_model = SimpleSamplingModel(likelihood=likelihood, prior=posterior) 
# uses SBI (e.g., ABC)
alt_second_posterior = SimpleSamplingModel(likelihood=likelihood, prior=posterior) 
```

### PR DESCRIPTION
**Pull request by jhuggins**
## Summary

- **Protocol-based conversion**: `converter_registry.convert(dist, SupportsLogProb)` now accepts `@runtime_checkable` protocols as targets, resolving to concrete types via registered resolver functions. This enables automatic conversion of sampling-only distributions (e.g., MCMC posteriors) into distributions that support density evaluation.
- **KDEDistribution**: New Gaussian KDE distribution backed by TFP `MixtureSameFamily`, satisfying `SupportsLogProb` + `SupportsSampling` + moment protocols. Provides a nonparametric alternative to moment-matching for protocol conversion.
- **WorkflowFunction integration**: `_convert_distributions()` now handles protocol type hints, auto-converting distributions that don't satisfy the required protocol.

## Architecture

Protocol resolution follows the existing `Converter` pattern rather than special-casing the registry. A new `ProtocolConverter` (priority 200) intercepts protocol targets, resolves them to concrete types via registered resolvers, and delegates back to the registry for the actual conversion. `ConverterRegistry` remains a simple dispatcher.

## Changes

- `probpipe/converters/_protocol_converter.py` – New `ProtocolConverter` class with `_is_protocol()`, resolvers, delegation to registry
- `probpipe/converters/__init__.py` – Register `ProtocolConverter` with `SupportsLogProb` and `SupportsSampling` resolvers
- `probpipe/converters/_probpipe.py` – `_convert_to_kde()` + dispatch table entry
- `probpipe/distributions/kde.py` – New `KDEDistribution` class
- `probpipe/core/node.py` – Protocol conversion in `_convert_distributions()`
- `probpipe/distributions/__init__.py` – Export `KDEDistribution`
- `probpipe/__init__.py` – Top-level export
- `tests/test_converters.py` – 25 new tests

## Test plan

- [x] `TestProtocolConversion` (9 tests): already-satisfied returns same, scalar/MV empirical conversion, check protocol, unregistered raises, from_distribution API, provenance
- [x] `TestKDEDistribution` (16 tests): construction, log_prob, sampling, moments, weights, bandwidth, protocols, conversions, repr
- [x] Full test suite: 1393 passed, 5 skipped, 0 failures

Closes #95

### CODE REVIEW

File: probpipe/converters/_registry.py [hunk 1]
```diff
@@ -38,16 +45,66 @@ class ConversionInfo:
...
+    """Choose Normal or MultivariateNormal based on event dimensionality."""
+    from ..distributions.continuous import Normal
+    from ..distributions.multivariate import MultivariateNormal
+
+    try:
+        event_shape = dist.event_shape
+    except AttributeError:
+        # If no event_shape, try dim
+        try:
+            if dist.dim <= 1:
```
Comment (arob5) [inline probpipe/converters/_registry.py]: When do we use `dim`? `PyTreeArrayDistribution` defines `event_size`, which seems like what we'd want here.

File: probpipe/distributions/kde.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        sample_var = jnp.einsum("n,nd->d", self._weights, diff ** 2)
+        return sample_var + self._bandwidth ** 2
+
+    def _cov(self) -> Array:
+        m = self._mean()
+        if self._scalar:
+            return self._variance()
+        flat = self._samples
+        diff = flat - m
+        sample_cov = jnp.einsum("ni,nj,n->ij", diff, diff, self._weights)
```
Comment (arob5) [inline probpipe/distributions/kde.py]: Weighed sample cov and sample var logic is likely duplicated in multiple points throughout the codebase. My thought is that this should be standardized in a utils file or something like this, perhaps in conjunction with standardizing the weight code: #98. This would avoid redundancy and clearly define the shape requirements/assumptions.
